pulp_smash/tests/pulp3/pulpcore/api_v3/test_tasks.py

```diff
@@ -0,0 +1,94 @@
+
+_DYNAMIC_TASKS_ATTRS = ('finished_at',)
+"""Task attributes that are dynamically set by Pulp, not set by a user."""
+
+
+class TasksTestCase(unittest.TestCase, utils.SmokeTest):
+    """Perform different operation over tasks.
+
+    This test targets the following issues:
+
+    * `Pulp #3144 <https://pulp.plan.io/issues/3144>`_
+    * `Pulp #3527 <https://pulp.plan.io/issues/3527>`_
+    * `Pulp Smash #754 <https://github.com/PulpQE/pulp-smash/issues/754>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class-wide variables."""
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.json_handler)
+        cls.client.request_kwargs['auth'] = get_auth()
+        cls.task = {}
+
+    def test_01_create_task(self):
+        """Create a task."""
+        repo = self.client.post(REPO_PATH, gen_repo())
+        self.addCleanup(self.client.delete, repo['_href'])
+        attrs = {'description': utils.uuid4()}
+        response = self.client.patch(repo['_href'], attrs)
+        self.task.update(self.client.get(response['_href']))
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_02_read_task(self):
+        """Read a task by its _href."""
+        task = self.client.get(self.task['_href'])
+        for key, val in self.task.items():
+            if key in _DYNAMIC_TASKS_ATTRS:
+                continue
+            with self.subTest(key=key):
+                self.assertEqual(task[key], val)
+
+    @selectors.skip_if(bool, 'task', False)
+    def test_03_read_tasks(self):
+        """Read a task by its worker."""
+        with self.subTest('invalid worker name'):
+            with self.assertRaises(HTTPError):
+                self.filter_tasks({'worker': utils.uuid4()})
+        with self.subTest('valid worker name'):
+            page = self.filter_tasks({'worker': self.task['worker']})
+            self.assertNotEqual(len(page['results']), 0)
+
+    def test_04_filter_invalid_date(self):
+        """Filter tasks by an invalid date."""
+        page = self.filter_tasks({
+            'finished_at': utils.uuid4(),
+            'started_at': utils.uuid4()})
+        self.assertEqual(len(page['results']), 0)
+
+    def test_05_cancel_tasks(self):
+        """Cancel tasks."""
+        href_canceled_tasks = []
+        for state in ('running', 'waiting'):
+            for task in self.filter_tasks({'state': state})['results']:
+                href_canceled_tasks.append(task['_href'])
+                self.client.post(urljoin(task['_href'], 'cancel/'))
+        for task_href in href_canceled_tasks:
+            self.assertEqual(self.client.get(task_href)['state'], 'canceled')
+
+    def test_06_delete_tasks(self):
+        """Delete tasks."""
+        href_deleted_tasks = []
+        for state in ('canceled', 'completed', 'failed', 'skipped'):
+            for task in self.filter_tasks({'state': state})['results']:
```

Comment:
Again, I worry about collateral damage. How about deleting just the task(s) spawned by this test method? This change in direction does mean less thorough test coverage. But I think the solution is to figure out how to make this test case better.